biodatasets/tmvar_v1/tmvar_v1.py

```diff
@@ -0,0 +1,240 @@
+}
+
+_SUPPORTED_TASKS = [Tasks.NAMED_ENTITY_RECOGNITION]
+
+_SOURCE_VERSION = "1.0.0"
+
+_BIGBIO_VERSION = "1.0.0"
+
+logger = datasets.utils.logging.get_logger(__name__)
+
+
+class tmvarV1Dataset(datasets.GeneratorBasedBuilder):
+    """
+    The tmVar dataset contains 500 PubMed articles manually annotated with mutation
+    mentions of various kinds.
+    It can be used for biomedical NER tasks
+    """
+
+    DEFAULT_CONFIG_NAME = "tmvar_v1_source"
+    SOURCE_VERSION = datasets.Version(_SOURCE_VERSION)
+    BIGBIO_VERSION = datasets.Version(_BIGBIO_VERSION)
+
+    BUILDER_CONFIGS = []
+    BUILDER_CONFIGS.append(
+        BigBioConfig(
+            name=f"{_DATASETNAME}_source",
+            version=SOURCE_VERSION,
+            description=f"{_DATASETNAME} source schema",
+            schema="source",
+            subset_id=f"{_DATASETNAME}",
+        )
+    )
+    BUILDER_CONFIGS.append(
+        BigBioConfig(
+            name=f"{_DATASETNAME}_bigbio_kb",
+            version=BIGBIO_VERSION,
+            description=f"{_DATASETNAME} BigBio schema",
+            schema="bigbio_kb",
+            subset_id=f"{_DATASETNAME}",
+        )
+    )
+
+    def _info(self) -> datasets.DatasetInfo:
+
+        if self.config.schema == "source":
+            features = datasets.Features(
+                {
+                    "pmid": datasets.Value("string"),
+                    "passages": [
+                        {
+                            "type": datasets.Value("string"),
+                            "text": datasets.Sequence(datasets.Value("string")),
+                            "offsets": datasets.Sequence([datasets.Value("int32")]),
+                        }
+                    ],
+                    "entities": [
+                        {
+                            "text": datasets.Sequence(datasets.Value("string")),
+                            "offsets": datasets.Sequence([datasets.Value("int32")]),
+                            "concept_id": datasets.Value("string"),
+                            "semantic_type_id": datasets.Sequence(
+                                datasets.Value("string")
+                            ),
+                        }
+                    ],
+                }
+            )
+        elif self.config.schema == "bigbio_kb":
+            features = schemas.kb_features
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager) -> List[datasets.SplitGenerator]:
+        """Returns SplitGenerators."""
+
+        url = _URLS[_DATASETNAME]
+        data_dir = dl_manager.download_and_extract(url)
+        train_filename = "tmVarCorpus\\train.PubTator.txt"
+        test_filename = "tmVarCorpus\\test.PubTator.txt"
```

Comment:
I don't think this syntax will be portable across different OSs. We can probably drop the `*_filename` variables and just build the path directly as the suggestion below

Reply:
Sorry, I only just saw this. Thank you for the changes